hail/src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -114,7 +90,7 @@ object IRFunctionRegistry {
       valueParameterTypes,
       IRParser.parseType(returnType),
       false,
-      (_, args, _) => Subst(body, BindingEnv(Env[IR](argNames.zip(args): _*))),
+      (_, args, _) => Subst(body, BindingEnv.eval(argNames.zip(args): _*)),
```

Comment:
This seemed like the only difference between `pyRegisterIR` and `pyRegisterIRForServiceBackend`. Is one preferred over the other?

Reply:
Nope, just aesthetics. I like this for being slightly more explicit.